artichoke-backend/src/extn/core/string/string_test.rb

```diff
@@ -182,4 +183,19 @@ def string_tr
   raise unless 'abcd'.tr('a-z', 'xxx') == 'xxxx'
 end
 
+def string_end_with
+  def assert_raise_type_error
```

Comment:
RuboCop doesn't seem to like this being nested. Feel free to move this helper to the top level. It reads really nicely this way with the block